src/model/StructureBase.h

```diff
@@ -126,22 +126,14 @@ class Structure : public StructureBase
     }
 
 protected:
-    template<std::size_t I = 0, typename ... Tp>
-    inline typename std::enable_if<I == (sizeof ... (Tp) - 1), void>::type
-    static toStringWithFields(const std::tuple<Tp ...> &t, const FieldNames &names, QTextStream &outStream)
+    template<class Tuple, std::size_t... Is>
+    static void toStringWithFields(const Tuple &t, std::index_sequence<Is...>, const FieldNames &names,
+                                   QTextStream &outStream)
     {
-        using TupleElementType = typename std::tuple_element<I, std::tuple<Tp ...> >::type;
-        outStream << names[I] << "=" << StringConversionHandler<TupleElementType>::toString(std::get<I>(t));
-    }
+        using expander = int[]; // workaround because fold expression is only available in C++17
 
-    template<std::size_t I = 0, typename ... Tp>
-    inline typename std::enable_if<(I < (sizeof ... (Tp) - 1)), void>::type
-    static toStringWithFields(const std::tuple<Tp ...> &t, const FieldNames &names, QTextStream &outStream)
-    {
-        using TupleElementType = typename std::tuple_element<I, std::tuple<Tp ...> >::type;
-        outStream << names[I] << "=" << StringConversionHandler<TupleElementType>::toString(std::get<I>(t));
-        outStream << ", ";
-        toStringWithFields<I + 1, Tp ...>(t, names, outStream);
+        (void)expander{0, (void(outStream << (Is == 0 ? "" : ", ") << names[Is]
```

Comment:
Please add missing "=" to the output:
 (void)expander{0, (void(outStream << (Is == 0 ? "" : ", ") << names[Is] << "="